crates/red_knot_python_semantic/resources/mdtest/protocols.md

```diff
@@ -259,6 +257,12 @@ isinstance(object(), RuntimeCheckableFoo)
 isinstance(object(), RuntimeCheckableBar)
 ```
 
+However, we understand that they are distinct symbols:
+
+```py
+static_assert(typing.Protocol is not typing_extensions.Protocol)
+```
```

Comment:
I suspect that there are other `KnownInstanceType` variants where we might currently incorrectly assume that the `typing_extensions` version of the symbol shares the same memory address as the `typing` version of the symbol, which would lead us to incorrect inferences in things like this